diff --git a/lib/default.nix b/lib/default.nix
index 751738a98965..b689f7744bac 100644
--- a/lib/default.nix
+++ b/lib/default.nix
@@ -48,6 +48,7 @@ let
       callLibs = file: import file { lib = self; };
     in
     {
+      patchedFunction = x: x;
 
       # often used, or depending on very little
       trivial = callLibs ./trivial.nix;
diff --git a/nixos/modules/module-list.nix b/nixos/modules/module-list.nix
index 6995f3ba1ed2..49adcce57c6f 100644
--- a/nixos/modules/module-list.nix
+++ b/nixos/modules/module-list.nix
@@ -2039,4 +2039,10 @@
       ./image/repart.nix
     ];
   }
+  ({ lib, config, ... }: {
+    options.patchedModule.test = lib.mkOption {
+      default = null;
+      example = "test";
+    };
+  })
 ]
diff --git a/pkgs/by-name/fl/flake-utils-plus-test/package.nix b/pkgs/by-name/fl/flake-utils-plus-test/package.nix
new file mode 100644
index 000000000000..ad5323375baf
--- /dev/null
+++ b/pkgs/by-name/fl/flake-utils-plus-test/package.nix
@@ -0,0 +1 @@
+{ coreutils }: coreutils
