diff --git a/lib/default.nix b/lib/default.nix
index 5a85c542117..0ed00136d1d 100644
--- a/lib/default.nix
+++ b/lib/default.nix
@@ -146,5 +146,7 @@ let
       nixType imap;
     inherit (self.versions)
       splitVersion;
+
+    patchedFunction = x: x;
   });
 in lib
diff --git a/nixos/modules/module-list.nix b/nixos/modules/module-list.nix
index c951db5822a..53226de942e 100644
--- a/nixos/modules/module-list.nix
+++ b/nixos/modules/module-list.nix
@@ -1191,4 +1191,10 @@
   ./virtualisation/vmware-guest.nix
   ./virtualisation/xen-dom0.nix
   ./virtualisation/xe-guest-utilities.nix
+  ({ lib, config, ... }: {
+    options.patchedModule.test = lib.mkOption {
+      default = null;
+      example = "test";
+    };
+  })
 ]
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index c904707b7cf..9f0cab1875d 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -12,6 +12,8 @@ with pkgs;
 
 {
 
+  flake-utils-plus-test = callPackage ../applications/misc/hello { };
+
   # A stdenv capable of building 32-bit binaries.  On x86_64-linux,
   # it uses GCC compiled with multilib support; on i686-linux, it's
   # just the plain stdenv.
